### PR TITLE
Fix issue with enforced timeout

### LINKED ISSUE #396: Customize subscription time
I'm running into an issue where a [webkit webview](https://developer.apple.com/documentation/webkit/wkwebview?language=objc) drops connections after 60s.  Pubnub, by default, listens for 280s. For periods where conversations may be idle when using the chat SDK, this causes a timeout. When interrupted, the SDK will kick up a `PNBadRequestCategory` error. If we attempt to reconnect the app starts falling apart unless we dispose of all the existing connections and re-connect to all existing channels.

Is there a way to customize this period? or any thoughts on a workaround?

My current workaround is to just broadcast a dead message into the users own channel with `signal` - but was hoping for something cleaner.

### PR DESCRIPTION
**Pull request by parfeon**
fix(transport): fix issue with enforced timeout

Fix issue because of which timeout enforced by browser triggered wrong error status category.

Closes #396

### CODE REVIEW

File: src/errors/pubnub-api-error.ts [hunk 1]
```diff
@@ -58,7 +58,9 @@ export class PubNubAPIError extends Error {
       category = StatusCategory.PNNetworkIssuesCategory;
       message = 'Network issues';
     } else if (errorName === 'TypeError') {
-      category = StatusCategory.PNBadRequestCategory;
+      if (message.indexOf('Load failed') !== -1 || message.indexOf('Failed to fetch') != -1)
```
Comment (mohitpubnub) [inline src/errors/pubnub-api-error.ts:61]: 👍🏻

**Review by mohitpubnub — APPROVED** [review]
LGTM!

File: src/errors/pubnub-api-error.ts (hunk 1 above)
Comment (mohitpubnub) [reply src/errors/pubnub-api-error.ts:61]: a dev to dev comment!
 What about using `includes()` instead of `indexOf`.

Comment (parfeon) [reply src/errors/pubnub-api-error.ts:61]: @mohitpubnub `fetch` errors not really talkative and don't provide any insights on what exact error it was (and looks like it is deliberately). So I've found errors which can be generated by `fetch` on macOS (`Load failed`) and win (apparently it is `Failed to fetch`)

**Comment by parfeon** [comment]
@pubnub-release-bot release

File: src/errors/pubnub-api-error.ts (hunk 1 above)
Comment (mohitpubnub) [reply src/errors/pubnub-api-error.ts:61]: Yeah!! It's looks a bit challenging to match more accurate error message clauses.!

**Pull request merged**